### PR TITLE
Table autodiscovery

### PR DESCRIPTION
**Pull request by mieciu**
Adding table auto discovery logic. It's pretty basic for now, but should elevate customer experience when they first set up Quesma.

### How it works
Auto-discovery kicks in whenever `indexes` section in configuration is left empty. Additionally to tables, we try to detect a "timestamp field" via simple heuristics.

### CODE REVIEW

File: quesma/clickhouse/schema_loader.go [hunk 1]
```diff
@@ -92,31 +96,55 @@ func (sl *tableDiscovery) ReloadTableDefinitions() {
...
 		return
 	} else {
-		for table, columns := range tables {
-			if indexConfig, found := sl.cfg.IndexConfig[table]; found {
-				if indexConfig.Enabled {
-					for colName := range columns {
-						if _, exists := indexConfig.Aliases[colName]; exists {
-							logger.Error().Msgf("column [%s] clashes with an existing alias, table [%s]", colName, table)
+		if sl.TableAutodiscoveryEnabled() {
+			var autoDiscoResults strings.Builder
```
Comment (pdelewski) [inline quesma/clickhouse/schema_loader.go:100]: Just for readability maybe we could extract separated functions for both branches

File: quesma/clickhouse/schema_loader.go [hunk 2]
```diff
@@ -92,31 +96,55 @@ func (sl *tableDiscovery) ReloadTableDefinitions() {
...
+			logger.Info().Msg("Index configuration empty, running table auto-discovery")
+			for table, columns := range tables {
+				comment := sl.SchemaManagement.tableComment(databaseName, table)
+				createTableQuery := sl.SchemaManagement.createTableQuery(databaseName, table)
+				var maybeTimestampField string
+				if sl.cfg.Hydrolix.IsNonEmpty() {
+					maybeTimestampField = sl.SchemaManagement.tableTimestampField(databaseName, table, "hydrolix")
```
Comment (pdelewski) [inline quesma/clickhouse/schema_loader.go:107]: I think that these strings `hydrolix` and `clickhouse` could be extracted as consts and used also in `tableTimestampField ` function

File: quesma/clickhouse/schema_loader.go (hunk 1 above)
Comment (mieciu) [reply quesma/clickhouse/schema_loader.go]: https://github.com/QuesmaOrg/quesma/pull/497/commits/d7af1dc3cd7e0cbfe3b501866a32681f24ddcde3

File: quesma/clickhouse/schema_loader.go (hunk 2 above)
Comment (mieciu) [reply quesma/clickhouse/schema_loader.go]: https://github.com/QuesmaOrg/quesma/pull/497/commits/c61928a18c8ba4d8398b4d8158299c9d229776fe

**Comment by pdelewski** [comment]
LGTM :)

**Review by pdelewski — APPROVED** [review]

**Pull request merged**